drag-and-drop.md

```diff
@@ -0,0 +1,780 @@
+LiveView version to 0.18. In `mix.exs` make sure you have:
+
+```elixir
+{:phoenix_live_view, "~> 0.18"}
+```
+
+While we are waiting for Phoenix 1.7 to be available we need to fix
+a breaking change linked to LiveView 0.18.
+The `live_flash/2` is now part of the [Phoenix.Component](https://hexdocs.pm/phoenix_live_view/0.18.3/Phoenix.Component.html#live_flash/2)
+module. This function will be deprecated with Phoenix 1.7 but to make
+sure our application can run we need to add this module in our `view_helper` function.
+
+In `app_web.ex` and `import Phoenix.Component` in the `view_helper` function.
+
+Before running the application we can clean the `lib/app_web/templates/layout/root.html.heex` file:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="csrf-token" content={csrf_token_value()} />
+    <script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
+    <%= live_title_tag(assigns[:page_title] || "App", suffix: " · Phoenix Framework") %>
+    <link phx-track-static rel="stylesheet" href={Routes.static_path(@conn, "/assets/app.css")} />
+    <script
+      defer
+      phx-track-static
+      type="text/javascript"
+      src={Routes.static_path(@conn, "/assets/app.js")}
+    >
+    </script>
+  </head>
+  <body>
+    <%= @inner_content %>
+  </body>
+</html>
+```
+
+Note that we have added `<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>`
+to the `head`. This will add Alpine.js features to our application.
+See the [Alpine.js documentation](https://alpinejs.dev/essentials/installation).
+
+If you prefer to avoid using the Alpine.js cdn link, you can download and save
+the content of the Alpine.js from https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js
+into `assets/vendor/alpine.js` file and import in `app.js` with:
+
+```js
+import Alpine from "../vendor/alpine"
+```
+
+
+
+And the `lib/app_web/templates/page/index.html.heex`:
+
+```html
+<.container>
+  <.h2 class="text-red-500">
+    Hello App!
+  </.h2>
+</.container>
+```
+
+You can now run `mix deps.get` to make sure all dependencies are installed
+and `mix phx.server`!
+
+If you'd like to have the formatter working for the `.heex`
+templates, you can update the `.formatter.exs` as describe in
+https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.HTMLFormatter.html
+
+There are a few steps to do for this setup.
+Hopefully this will be simplified with Phoenix 1.7 coming soon.
+Don't hesitate to open an issue on this Github repository if
+you still think there is some missing information.
+
+
+## Create items
+
+We can use the [mix phx.gen.live](https://hexdocs.pm/phoenix/Mix.Tasks.Phx.Gen.Live.html)
+command:
+
+run `mix phx.gen.live Tasks Item items text:string index:integer`
```

Comment:
I'm following this doc. When I was installing the Petal Components package, I noticed one of [their setup steps](https://petal.build/components) was 

```
You may have a modal/1 function defined already, which will cause a clash with PetalComponents.Modal.modal/1. Check in live_helpers.ex and remove it.
```

This only makes sense _after_ running this command. This might lead to confusion and a improper Petal set up for people that never did it 😥

Reply:
In addition to this, I've reached this stage and, even after following Petal's guide and running this command, I can't seem to get a fresh project running. I get a 

```
** (CompileError) lib/app_web/views/page_view.ex: undefined function container/1 (expected AppWeb.PageView to define such a function or for it to be imported, but none are available)
```

Perhaps I'm missing something but I'm struggling to get it running 😞 . Perhaps it's me but I feel like something is missing here 👀